### DISCUSSION TITLE
PhiAccrualFailureDetector now needs a new param

### DISCUSSION THREAD

**Discussion by aley1** [dotnet/dotNext#206] [Q&A]
Hi Sakno,

Before upgrading to version 4.5, I was using the PhiAccrualFailureDetector as the failure detection mechanism
var cluster = new RaftCluster(configuration)
    {
        FailureDetectorFactory = static member => new PhiAccrualFailureDetector() { Threshold = 3D },
    };

Now it doesn't work - it looks a new param (estimatedFirstHeartbeat) is needed.
Could you please kindly explain what are the implications / use of this param?
And I tried initiating it with a new TimeSpan but doesn't work.

**Comment by sakno** [comment]
`estimatedFirstHeartbeat` is required to initialize internal reservoir of samples. It is presented in all other implementations. Having this class without `estimatedFirstHeartbeat` is a result of bad design from my side. This is why it was modified. `estimatedFirstHeartbeat` represents a guess about response time of the first heartbeat message produced by the leader. You can set it to lower election timeout optimistically.
